environment-upstream-dev.yml

```diff
@@ -0,0 +1,51 @@
+name: notebook-gallery
+channels:
+- conda-forge
+- defaults
+dependencies:
+- bokeh
+- bottleneck
+- cartopy
+- cf_units
+- cftime>=1.0.4.2
+- cmaps
+- cmocean
+- dask-jobqueue>=0.7.0
+- dask>=2.7
+- eofs
+- h5py
+- holoviews
+- hvplot
+- ipywidgets
+- jupyterlab
+- matplotlib
+- metpy
+- mpi4py
+- nbval
+- netcdf4
+- nodejs
+- numba
+- pandas
+- pandoc
+- pandocfilters
+- pip
+- pre-commit
+- psutil
+- pytest
+- rasterio
+- scikit-learn
+- scipy
+- seaborn
+- statsmodels
+- tqdm
+- watermark
+- xesmf
+- zarr
+- pip:
+  - git+https://github.com/pydata/xarray.git
+  - git+https://github.com/intake/intake.git
+  - git+https://github.com/NCAR/intake-esm.git
+  - git+https://github.com/NCAR/esmlab.git
+  - git+https://github.com/NCAR/notebook-gallery.git#egg=ngallery_utils
```

Comment:
Isn't it the case that this will always install the master branch of `notebook-gallery`? What happens when the PR includes changes to the `notebook-gallery` modules? I think we want to install this differently i.e. after the environment creation/update step in `.circleci/config.yml`